plugins/modules/s3_object.py

```diff
@@ -636,6 +647,35 @@ def path_check(path):
         return False
 
 
+def get_content_type(src, present=True):
+    if not present:
+        content_type = None
+        if src:
+            content_type = mimetypes.guess_type(src)[0]
+        if content_type is None:
+            # s3 default content type
+            content_type = 'binary/octet-stream'
+        return content_type
+
+
+def get_extra_params(module, encrypt, metadata=None):
```

Comment:
You pass the whole `module` object just to access `module.params['encryption_mode']` and `module.params['encryption_kms_key_id']`. I think it would be more elegant to just pass these 2 keys through the function arguments. This way you've got a better control on your function inputs.